test/unit/src/common/components/repositories-home/t_repositories-home.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import expect from 'expect';
+import { shallow } from 'enzyme';
+import proxyquire from 'proxyquire';
+import sinon from 'sinon';
+
+const RepositoriesHome = proxyquire(
+  '../../../../../../src/common/components/repositories-home',
+  {
+    '../../actions/snaps': {
+      fetchUserSnaps: sinon.stub().returns({ type: 'FETCH_SNAP_REPOSITORIES' })
```

Comment:
I guess that's where `mapDispatchToProps` would help as suggested by @earnubs.

We would be able to pass mocked dispatch calls to raw component via props without ugly proxyquire calls.